mypy/semanal.py

```diff
@@ -5721,9 +5713,7 @@ def get_module_symbol(self, node: MypyFile, name: str) -> SymbolTableNode | None
                 sym = SymbolTableNode(GDEF, self.modules[fullname])
             elif self.is_incomplete_namespace(module):
                 self.record_incomplete_ref()
-            elif "__getattr__" in names and (
-                node.is_stub or self.options.python_version >= (3, 7)
-            ):
+            elif "__getattr__" in names and node.is_stub:
```

Comment:
Previously we recognised module-level `__getattr__` methods if it was a stub _or_ `--python-version` was set to >=3.7. Your PR currently would mean we'd now _only_ recognise module-level `__getattr__` methods in stubs, regardless of what `--python-version` was set to!

```suggestion
            elif "__getattr__" in names:
```

Reply:
Looks like you beat me to it ;)